core/src/main/java/com/linecorp/armeria/server/http/annotation/ParamMap.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.annotation;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Pre-defined interface type for the whole request parameter names and values.
+ */
+public interface ParamMap extends Map<String, List<String>> {
```

Comment:
I'm curious if we could extend `io.netty.codec.Headers<String, String, ...>`, which is designed for a multi-value dictionary from the beginning. Please take a look into `DefaultHttpHeaders` in Armeria as a reference.